File: BinarySearch/Median of 2 Sorted Arrays of Different Sizes.java

class Solution {
    public double medianOf2(int a[], int b[]) {
        // Merge both arrays
        int[] c = new int[a.length + b.length];
        int index = 0;
        
        for (int i = 0; i < a.length; i++) {
            c[index++] = a[i];
        }
        for (int i = 0; i < b.length; i++) {
            c[index++] = b[i];
        }

        // Sort the merged array
        Arrays.sort(c);

        int n = c.length;
        if (n % 2 == 0) {
            return (c[n / 2 - 1] + c[n / 2]) / 2.0;
        } else {
            return c[n / 2];
        }
    }
}

File: BinarySearch/Median of 2 Sorted Arrays of Different Sizes.js
/**
 * @param {Number[]} arr1
 * @param {Number[]} arr2
 * @returns {Number}
 */
class Solution {
    medianOf2(a, b) {
        // Merge both arrays
        let c = a.concat(b);

        // Sort using a numeric comparator
        c.sort((x, y) => x - y);

        let n = c.length;
        if (n % 2 === 0) {
            // If even, return average of the two middle numbers
            return (c[n / 2 - 1] + c[n / 2]) / 2;
        } else {
            // If odd, return the middle number
            return c[Math.floor(n / 2)];
        }
    }
}
